github/GithubRetry.py

```diff
@@ -46,6 +46,8 @@
 from github.Requester import Requester
 
 DEFAULT_SECONDARY_RATE_WAIT: int = 60
+DEFAULT_STATUS_FORCELIST = list(range(500, 600))
```

Comment:
I think we should also add `403` to the default.
```suggestion
DEFAULT_STATUS_FORCELIST = list(range(500, 600)) + [403]
```

Reply:
@jodelasur Any objections? Otherwise I would apply these changes.

Reply:
I guess this is OK. See my [comment below](https://github.com/PyGithub/PyGithub/pull/3006#discussion_r1818275896) though. May we could apply this and not the other one?